src/coreComponents/physicsSolvers/fluidFlow/ReactiveCompositionalMultiphaseOBL.cpp

```diff
@@ -161,6 +161,21 @@ void ReactiveCompositionalMultiphaseOBL::implicitStepComplete( real64 const & ti
                                                 [&]( localIndex const,
                                                      ElementSubRegionBase & subRegion )
     {
+      // for output purposes (visualization, etc) we update the last component fraction
```

Comment:
Actually, can you be more precise in this comment about what you actually do?
It looks like you enforce the sum to 1?
You could also have divided every component by the sum? Why only modifying the last slot?

Reply:
We have to update the last component fraction only because it is never updated by `ApplySystemSolution` here:
https://github.com/GEOSX/GEOSX/blob/51a10972167b388766c2b130bfd99560a6f12af9/src/coreComponents/physicsSolvers/fluidFlow/ReactiveCompositionalMultiphaseOBL.cpp#L487-L493
In fact, this last component fraction is never used during the simulation and the purpose of this change is just to make sure that the output component fractions (say, in VTK or Time History) have the correct values and sum to one. All the other component fractions have already been updated by the Newton update in `ApplySystemSolution` and should not be modified at this point.